### ISSUE TITLE
Make dragging unselected POI do "select & pan"

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#3562]
**[Submitted to the original trac issue database at 2.25pm, Thursday, 3rd March 2011]**

(Richard Mann, potlatch-dev 4 mar 2011)
"I don't know whether it's really a bug or not, but I do find myself
moving objects unintentionally when panning, which isn't entirely
user-friendly. I'd prefer that drag on an unselected object resulted
in select & pan."

Fwiw, I agree.

**Comment by tomhughes** [comment]
**Author: bryce2@obviously.com**
**[Added to the original trac issue at 2.11am, Wednesday, 8th June 2011]**

See also: http://trac.openstreetmap.org/ticket/3810
I also find myself moving objects unintentionally.
And even more often wondering if I just did move something or not (but afraid to press undo because it might undo a real change I just made).
-------------------------------------------------------

The options here seem to be:
1) Drag unselected object moves the object.
2) Drag unselected object results in select & pan.
3) Beginners pan with arrow keys, advanced users pan with shift-drag
4) Use undocumented ActionScript to allow advanced users to drag with the right mouse button (ala JOSM).

Personally I like option #3, even though it presents some barrier to new users understanding how to move around.

**Comment by tomhughes** [comment]
**Author: stevage**
**[Added to the original trac issue at 10.40pm, Wednesday, 8th June 2011]**

I like 2. Panning is 10 (100? 1000?) times more frequent than dragging, so having the modifier for panning is just wrong. Also, having shift+drag=pan means we couldn't have selection boxes.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 10.41pm, Wednesday, 8th June 2011]**

2 is clearly the only sane one. :)

**Comment by tomhughes** [comment]
**Author: stevage**
**[Added to the original trac issue at 1.02am, Monday, 4th July 2011]**

Ugh, if evidence was required that ways get dragged accidentally:

http://www.openstreetmap.org/browse/way/30142120/history

Just discovered this - apparently I dragged a street a few hundred metres to the west, and saved it, without even noticing.

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 4.29pm, Monday, 18th July 2011]**

Dragging an object now requires the object to be selected first. Panning behaviour remains unchanged, i.e. you have to drag on an empty bit of map. I'll see how this pans out (pun unintended) in the medium term, and then maybe consider adding the ability to pan if you drag an unselected object, but for now I'd rather not.

Thought for the day: the fix was deleting three lines and changing half a line of code (https://github.com/systemed/potlatch2/commit/400350561611374b406a1bd91bb8474f2f4d070e), and took about five minutes. People have spent orders of magnitude more time filing tickets/usability reports/what-have-you about this... than actually fixing it. If only people were more prepared to roll up their sleeves and do stuff... ;)

**Comment by tomhughes** [comment]
**Author: stevage**
**[Added to the original trac issue at 3.43am, Friday, 9th September 2011]**

A usability issue I have noticed:
1) User "drags" unselected POI
2) While mouse is over POI, it becomes selected, but as mouse moves further, it becomes unselected again
3) User releases mouse. POI is unselected, and has not moved.
4) Repeat from step 1.

TL;DR: dragging an unselected POI should select the POI.
